Kitodo/src/main/java/org/kitodo/production/migration/NewspaperProcessesMigrator.java

```diff
@@ -619,6 +619,22 @@ public void createNextYearProcess() throws ProcessGenerationException, IOExcepti
         }
     }
 
+    private void createWorkpiece(Entry<String, IncludedStructuralElement> yearToCreate, String yearTitle, Process yearProcess) throws IOException {
+        Workpiece yearWorkpiece = new Workpiece();
+        yearWorkpiece.setId(yearProcess.getId().toString());
+        yearWorkpiece.setRootElement(yearToCreate.getValue());
+        RulesetManagementInterface ruleset = ServiceManager.getRulesetService().openRuleset(overallProcess.getRuleset());
```

Comment:
Please do not parse the ruleset file every time. It is already opened in function `initializeMigrator()`. You better change the local variables `rulesetManagement` and `newspaperView` there (line 284, 286) into class fields, and just access them here.

Reply:
This is right for the ruleset, but not for the newspaper, because I explicitly need the yearView.